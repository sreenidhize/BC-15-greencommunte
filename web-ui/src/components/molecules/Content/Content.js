import React from 'react';
import {makeStyles, Typography} from '@material-ui/core';
import PropTypes from "prop-types";

const useStyles = makeStyles((theme) => ({
    list:{
        paddingLeft: theme.spacing(4),
        margin: 0
    }
}));

const Content = ({heading, description}) => {
    const classes = useStyles();

    return (
        <div data-testid="job-content">
            <Typography variant={'h2'}>{heading}</Typography>
            {
                description.length && description.length > 1 ?
                    <ul className={classes.list}>
                        {
                            description.map((item,index) => <li key={index}>
                                <Typography variant={'body2'}>{item}</Typography>
                            </li>)
                        }
                    </ul>
                    : <Typography variant={'body2'}>{description[0]}</Typography>
            }
        </div>
    );
};


Content.propTypes = {
    heading: PropTypes.string.isRequired,
    description: PropTypes.array.isRequired,
};

export default Content;