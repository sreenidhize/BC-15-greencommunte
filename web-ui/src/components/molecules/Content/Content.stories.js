import React from 'react';
import Content from "./Content";
import theme from '../../../utils/theme/theme';
import { ThemeProvider } from '@material-ui/core';

export default {
    title: 'Molecules/Content',
    component: Content
};

const Template = (args) => <ThemeProvider theme={theme}><Content {...args} /></ThemeProvider>;

export const NormalContent = Template.bind({});
NormalContent.args = {
    heading: 'Description',
    description: ["Open Text is seeking a talented, personable interaction designer who can assist the User Experience Design team by working with other designers and development teams on a variety of projects. The OpenText User Experience Design group is a distributed multi-disciplinary team of professionals that are responsible for enhancing the UX of the companyâ€™s collective product suites worldwide."]
};

export const ListContent = Template.bind({});
ListContent.args = {
    heading: 'What it takes',
    description: ["High level of proficiency with leading UX Design software packages, such as Axure, Sketch, InVision", " Experience Design including the core Adobe Creative Suite products."]
};
