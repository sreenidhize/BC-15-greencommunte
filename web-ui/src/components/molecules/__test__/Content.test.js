import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Content from "../Content/Content";

describe("Content", () => {
    it("Matches the snapshot", () => {
        const { container } = render(
            <Content heading={"Description"} description={["test"]}/>
        );
        expect(container).toMatchSnapshot();
    });

    test("test content render", () => {
        const { getByTestId } = render(<Content heading={"Description"} description={["test"]}/>);
        expect(getByTestId("job-content")).toBeInTheDocument();
    });
});