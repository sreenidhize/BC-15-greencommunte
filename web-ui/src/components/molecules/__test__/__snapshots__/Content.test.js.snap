// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Matches the snapshot 1`] = `
<div>
  <div
    data-testid="job-content"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2"
    >
      Description
    </h2>
    <p
      class="MuiTypography-root MuiTypography-body2"
    >
      test
    </p>
  </div>
</div>
`;
